### PR TITLE
Use IPInfo as secondary source

### PR DESCRIPTION
**Pull request by enoch85**
Fix https://github.com/nextcloud/vm/issues/2682

### CODE REVIEW

**Comment by enoch85** [comment]
Hmm, this doesn't seem to work. Allowing "SE" in the list still blocks it when trying to reach the site.

@abdullahdevrel Are your `mmdb` different from Maxminds?

**Comment by enoch85** [comment]
Example from CSV:

```
start_ip	end_ip	country	country_name	continent	continent_name
1.0.0.0	1.0.0.255	AU	Australia	OC	Oceania
1.0.1.0	1.0.3.255	CN	China	AS	Asia
1.0.4.0	1.0.7.255	AU	Australia	OC	Oceania
1.0.8.0	1.0.15.255	CN	China	AS	Asia
1.0.16.0	1.0.31.255	JP	Japan	AS	Asia
```

In the Apache conf, we use different "tags". This might not work after all?

**Comment by enoch85** [comment]
Found this now: https://community.ipinfo.io/t/migrating-from-maxmind-geolite2-country-to-ipinfo-country-mmdb-edition/56

**Comment by enoch85** [comment]
So, since IPInfo doesn't run it's own Apache module, I'm afraid this won't work. :|

**Comment by abdullahdevrel** [comment]
@enoch85 Thank you for reviewing. The scheme is different from MaxMind's, and you have found the post that I would have linked. 

I believe the Apache module (https://github.com/maxmind/mod_maxminddb) is schema agnostic, like all the other MMDB reader libraries out there (https://community.ipinfo.io/t/list-of-mmdb-reader-libraries/2821).

Looking at this bit of code:

https://github.com/nextcloud/vm/blob/6a111c790e6d0207a5778d14c71af26bbf3a100e/network/geoblock.sh#L224-L227

So, with respect to our database schema, I assume this will be:

```
  MaxMindDBEnv MM_CONTINENT_CODE DB/continent
  MaxMindDBEnv MM_COUNTRY_CODE DB/country
```

Can you see if it would work?

**Comment by enoch85** [comment]
> @enoch85 Thank you for reviewing. The scheme is different from MaxMind's, and you have found the post that I would have linked.
> 
> I believe the Apache module (https://github.com/maxmind/mod_maxminddb) is schema agnostic, like all the other MMDB reader libraries out there (https://community.ipinfo.io/t/list-of-mmdb-reader-libraries/2821).
> 
> Looking at this bit of code:
> 
> https://github.com/nextcloud/vm/blob/6a111c790e6d0207a5778d14c71af26bbf3a100e/network/geoblock.sh#L224-L227
> 
> So, with respect to our database schema, I assume this will be:
> 
> ```
>   MaxMindDBEnv MM_CONTINENT_CODE DB/continent
>   MaxMindDBEnv MM_COUNTRY_CODE DB/country
> ```
> 
> Can you see if it would work?

Hmm, even if it would work it would mean we would have to maintain (or at least put checks in) to know which DB is used. So that would also mean that we would have to choose from either IPInfo or Maxmind. 

I will test anyway so that we'll know if that's the case.

**Comment by enoch85** [comment]
And yes, I can confirm, the change works.

Now I need to get my head around how we want to do this...

**Comment by enoch85** [comment]
Found a solution!

```
<IfModule mod_maxminddb.c>
  MaxMindDBEnable On

  # Check for IPinfo mmdb
  <IfFile "/usr/share/GeoIP/IPInfo-Country.mmdb">
    MaxMindDBFile DB /usr/share/GeoIP/IPInfo-Country.mmdb
    MaxMindDBEnv MM_CONTINENT_CODE DB/continent
    MaxMindDBEnv MM_COUNTRY_CODE DB/country
  </IfFile>
  # Check for Maxmind mmdb
  <IfFile "/usr/share/GeoIP/GeoLite2-Country.mmdb">
    MaxMindDBFile DB /usr/share/GeoIP/GeoLite2-Country.mmdb
    MaxMindDBEnv MM_CONTINENT_CODE DB/continent/code
    MaxMindDBEnv MM_COUNTRY_CODE DB/country/iso_code
  </IfFile>
</IfModule>
```

**Comment by enoch85** [comment]
OK, so this is tested and works!

@abdullahdevrel Please confirm that the  "attribution requirements" are enough.

File: lib.sh [hunk 1]
```diff
@@ -414,18 +416,29 @@ download_geoip_mmdb() {
...
         docker rm -f maxmind
         rm -f /tmp/dockerenv
+        # Since only one mmdb file can exist at the same time due to Apache "if" confitions, remove IPInfos config
+        rm -f "$GEOBLOCK_DIR"/IPInfo-Country.mmdb
     else
         docker rm -f maxmind
         rm -f /tmp/dockerenv
-        msg_box "Update limit for Maxmind GeoDatabase reached! Please try again tomorrow."
-        return 1
+        print_text_in_color "$ICyan" "Rate limit for Maxmind GeoDatabase reached! We're now trying to get the Country Database from https://ipinfo.io instead."
```
Comment (enoch85) [inline lib.sh:430]: @abdullahdevrel This is what I need approval on.

Comment (enoch85) [reply lib.sh:430]: I'll merge this now, please come back with a suggestion if the above is not enough.

**Pull request merged**